config/doxygen_config.in

```diff
@@ -2095,7 +2095,10 @@ INCLUDE_FILE_PATTERNS  =
 # recursively expanded use the := operator instead of the = operator.
 # This tag requires that the tag ENABLE_PREPROCESSING is set to YES.
 
-PREDEFINED             = "Remember(c)=c"
+# https://www.doxygen.nl/manual/preprocessing.html
+
+PREDEFINED             = "Remember(c)=c" \
+                         "C4_MPI=1"
```

Comment:
This extra "C4_MPI=1" line just causes the doxygen to be generated for the MPI case (rather than serial), right? That makes sense :)

Reply:
Right.  That forces Doxygen to process all `c4` packages with `#define C4_MPI 1`.  This is needed to allow Doxygen to generate the correct file linkage in the docs and diagrams.